### DISCUSSION TITLE
When building a component to publish to NPM, how can I export my css as a single file?

### DISCUSSION THREAD

**Discussion by vedantroy** [tailwindlabs/tailwindcss#11376] [Help]
I have a component that I want to publish as an NPM package. This component is using tailwind CSS classes.

Is there a way for me to build this component such people can just do:
```
import MyComponent from "component-lib"

export default function() {
   return <MyComponent />
}
```

and the CSS just works?

**Comment by palonza** [comment]
Hi @vedantroy, 

Hello friend, I hope I can help you and that it is not too late... if it helps you, do not forget to mark your question as answered, please.

### Requirements
- Webpack (to package all resources in one)
- NPM account

### Steps

Broadly speaking, you will need the following, make sure you have a 'tailwind.config.js' if not, run this command:

```bash
npx tailwindcss init
```

Create a CSS file 'style.css' and import the tailwind css styles, like this (This will import all the base, component, and utility styles provided by Tailwind CSS):

```css
@import 'tailwindcss/base';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
```

Then, run this:

```bash
webpack --mode production
```

Navigate to the root directory of your component library project, where the 'package.json' file is located, run the following command to create a new version of your component library:

```bash
npm version <version_number>
```

Replace <version_number> with the appropriate version number according to semantic versioning (e.g., patch, minor, or major), or specify a specific version number (e.g., 1.0.0).

Next, run the build command to ensure your library is up to date:

```bash
npm run build
```

Obviously, should you have an account on the npm registry (https://www.npmjs.com/), then, this command will trigger the build process to generate the bundled version of your component library:

```bash
npm publish
```

You will also need to be informed about the conditions, requirements, and other, check this:
https://docs.npmjs.com

Regards, palonza

**Reply by SeppeMarien** [reply]
Hey @palonza, I have been trying the same thing. I have setup my project like you suggested but still no luck. I have been wondering where do you import your  style.css?

**Reply by palonza** [reply]
Hi @SeppeMarien, in the context of publish component in npm repository or develop project too, when you run webpack --mode production, webpack is responsible to build and compress your .css in one file.
